cfg/optimize_test.go

```diff
@@ -0,0 +1,448 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+package cfg
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// Mock IsValueSet for testing.
+type mockIsValueSet struct {
+	setFlags    map[string]bool
+	boolFlags   map[string]bool
+	stringFlags map[string]string
+}
+
+func (m *mockIsValueSet) IsValueSet(flag string) bool {
+	return m.setFlags[flag]
+}
+
+func (m *mockIsValueSet) IsSet(flag string) bool {
+	return m.setFlags[flag]
+}
+
+func (m *mockIsValueSet) GetBool(flag string) bool {
+	return m.boolFlags[flag]
+}
+
+func (m *mockIsValueSet) GetString(flag string) string {
+	return m.stringFlags[flag]
+}
+
+func (m *mockIsValueSet) Set(flag string) {
+	m.setFlags[flag] = true
+}
+
+func (m *mockIsValueSet) SetString(flag string, value string) {
+	m.stringFlags[flag] = value
+}
+
+func (m *mockIsValueSet) SetBool(flag string, value bool) {
+	m.boolFlags[flag] = value
+}
+
+func (m *mockIsValueSet) Unset(flag string) {
+	delete(m.setFlags, flag)
+}
+
+// Helper function to create a test server.
+func createTestServer(t *testing.T, handler http.HandlerFunc) *httptest.Server {
+	t.Helper()
+	server := httptest.NewServer(handler)
+	return server
+}
+
+// Helper function to close a test server.
+func closeTestServer(t *testing.T, server *httptest.Server) {
+	t.Helper()
+	server.Close()
+}
+
+// Helper function to reset metadataEndpoints.
+func resetMetadataEndpoints(t *testing.T) {
+	t.Helper()
+	metadataEndpoints = []string{
+		"http://metadata.google.internal/computeMetadata/v1/instance/machine-type",
+	}
+}
+
+func TestGetMachineType_Success(t *testing.T) {
+	resetMetadataEndpoints(t)
+	// Create a test server that returns a machine type.
+	server := createTestServer(t, func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Metadata-Flavor", "Google")
+		fmt.Fprint(w, "zones/us-central1-a/machineTypes/n1-standard-1")
+	})
+	defer closeTestServer(t, server)
+
+	// Override metadataEndpoints for testing.
+	metadataEndpoints = []string{server.URL}
+
+	machineType, err := getMachineType(&mockIsValueSet{})
+	if !assert.NoError(t, err) {
+		return
+	}
+	assert.Equal(t, "n1-standard-1", machineType)
+}
```

Comment:
The test should have three parts - Arrange, Act and Assert and they should be separated by blank lines. Basically, here everything before the getMachineType is Arrange and there should not be blank lines in between. Everything after getMachineType call is in Assert. The getMachineType call is the "Act". So, please add a blank line after Act block. 

Ditto for other test methods